src/rules/linalg/factorization.jl

```diff
@@ -79,3 +82,192 @@ function _add!(X::AbstractMatrix{T}, Y::AbstractMatrix{T}) where T<:Real
     end
     X
 end
+
+#####
+##### `cholesky`
+#####
+
+function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
+    F = cholesky(X)
+    ∂X = Rule(Ȳ->chol_blocked_rev(Matrix(Ȳ), Matrix(F.U), 25, true))
+    return F, ∂X
+end
+
+function rrule(::typeof(getproperty), F::Cholesky, x::Symbol)
+    if x === :U
+        if F.uplo === 'U'
+            ∂F = Ȳ->UpperTriangular(Ȳ)
+        else
+            ∂F = Ȳ->LowerTriangular(Ȳ')
+        end
+    elseif x === :L
+        if F.uplo === 'L'
+            ∂F = Ȳ->LowerTriangular(Ȳ)
+        else
+            ∂F = Ȳ->UpperTriangular(Ȳ')
+        end
+    end
+    return getproperty(F, x), (∂F, DNERule())
+end
+
+# TODO: The comment below about row- versus column-major is incorrect; Julia's arrays are
+# column-major. This comment along with the below implementation was copied from Nabla and
+# works, but the implementation should perhaps should be revisited to ensure we're actually
+# being cache friendly.
+#
+# See [1] for implementation details: pages 5-9 in particular. The derivations presented in
+# [1] assume column-major layout, whereas Julia primarily uses row-major. We therefore
+# implement both the derivations in [1] and their transpose, which is more appropriate to
+# Julia.
```

Comment:
Canwe **not** tranpose comments we know are wrong?
Can we just open an issue about checking this for cache friendlyness and delete 80% of this comment?

Reply:
I didn't analyze the algorithm in depth to determine whether the code itself is the transpose of what it should be, but I figured that comment would be a good reminder to go back and fact check. An issue would work as well I suppose.

@willtebbutt, would you be willing to take a look at this, since you're the original author of the comment?

Reply:
Happy to follow up with this at some point if @oxinabox 's suggestion is followed. Certainly it would be useful to get rid of the comment for now.